abaco.go

```diff
@@ -201,27 +204,36 @@ func (group *AbacoGroup) demuxData(datacopies [][]RawType, frames int) int {
 		if frameAvail > frames {
 			break
 		}
-
 		frames -= frameAvail
 		packetsConsumed++
+
 		switch d := p.Data.(type) {
 		case []int16:
-			// Reading vector d in order was faster than the reverse.
-			for j, val := range d {
-				idx := j % group.nchan
-				datacopies[idx] = append(datacopies[idx], RawType(val))
+			// Reading vector d in channel order was faster than in packet-data order.
+			nsamp := len(d)/nchan
```

Comment:
you can avoid this division by checking j against length d

Reply:
Yes, but that would only delay it. I still need `nsamp` at the end of the loop to advance `samplesConsumed`. And in your way, I'd need a new variable to store `len(d)` so that we don't have to call it as a function once per loop iteration. So while I don't insist that my way is better, I don't believe yours is, either.